### ISSUE TITLE
Fix S1144 FP: Should not raise for abstract methods

### ISSUE THREAD

**Issue by pavel-mikula-sonarsource** [SonarSource/sonar-dotnet#4918]
Abstract `DoSomething` is used as a contract and overriden => used.

We should not raise on abstract members. They are either used by derived class, or the whole base class should be removed (different rule)
```C#
    public class Repro
    {
        public void Usage()
        {
            var usage = new Implementation();
            usage.DoSomething();
        }

        private abstract class Base
        {
            public abstract void DoSomething(); // Noncompliant FP
        }

        private class Implementation : Base
        {
            public override void DoSomething() => throw new System.NotImplementedException();
        }
    }
```

**Comment by pavel-mikula-sonarsource** [comment]
Or maybe not. If it's used only by inherited class, it doesn't need to be abstract in the base
